### ISSUE TITLE
JEC 2018 need  database file

### ISSUE THREAD

**Issue by bortigno** [UFLX2MuMu/Ntupliser#60]
2018 Data and MC JEC are not in a GT yet. Need to load them directly from the database file. See https://twiki.cern.ch/twiki/bin/view/CMS/JECDataMC#Recommended_for_MC

**Comment by bortigno** [comment]
I already downloaded the files in data/JEC on my local repo but didn't add them to version control as I am waiting for the PR to get merged before moving forward.

**Comment by bortigno** [comment]
Started working on it from branch issue-60-jec-2018-from-db

**Comment by bortigno** [comment]
In order to learn how to use a db file for the JEC I looked in https://twiki.cern.ch/twiki/bin/view/CMSPublic/WorkBookJetEnergyCorrections#JecSqliteFile
and added the configuration to the test cfg. In order to know which record, tag, and label to pass to the config I did:
```
conddb --db ../data/JEC/Autumn18_V8_MC.db listTags
[2019-05-15 11:13:56,005] INFO: Connecting to ../data/JEC/Autumn18_V8_MC.db [sqlite:///../data/JEC/Autumn18_V8_MC.db]
Name                                                        TimeType  ObjectType                        Synchronisation  EndOfValidity  Insertion_time              Description  
----------------------------------------------------------  --------  --------------------------------  ---------------  -------------  --------------------------  -----------  
JetCorrectorParametersCollection_Autumn18_V8_MC_AK4PF       Run       JetCorrectorParametersCollection  any              -1             2019-03-06 17:15:48.153879  New Tag      
JetCorrectorParametersCollection_Autumn18_V8_MC_AK4PFchs    Run       JetCorrectorParametersCollection  any              -1             2019-03-06 17:15:48.179636  New Tag      
JetCorrectorParametersCollection_Autumn18_V8_MC_AK4PFPuppi  Run       JetCorrectorParametersCollection  any              -1             2019-03-06 17:15:48.202311  New Tag      
JetCorrectorParametersCollection_Autumn18_V8_MC_AK4Calo     Run       JetCorrectorParametersCollection  any              -1             2019-03-06 17:15:48.212816  New Tag      
JetCorrectorParametersCollection_Autumn18_V8_MC_AK4JPT      Run       JetCorrectorParametersCollection  any              -1             2019-03-06 17:15:48.220884  New Tag      
JetCorrectorParametersCollection_Autumn18_V8_MC_AK8PF       Run       JetCorrectorParametersCollection  any              -1             2019-03-06 17:15:48.229330  New Tag      
JetCorrectorParametersCollection_Autumn18_V8_MC_AK8PFchs    Run       JetCorrectorParametersCollection  any              -1             2019-03-06 17:15:48.257512  New Tag      
JetCorrectorParametersCollection_Autumn18_V8_MC_AK8PFPuppi  Run       JetCorrectorParametersCollection  any              -1             2019-03-06 17:15:48.282349  New Tag 
```

and for the data
```
conddb --db ../data/JEC/Autumn18_RunABCD_V8_DATA.db listTags
[2019-05-15 11:25:13,290] INFO: Connecting to ../data/JEC/Autumn18_RunABCD_V8_DATA.db [sqlite:///../data/JEC/Autumn18_RunABCD_V8_DATA.db]
Name                                                                  TimeType  ObjectType                        Synchronisation  EndOfValidity  Insertion_time              Description                                                                                              
--------------------------------------------------------------------  --------  --------------------------------  ---------------  -------------  --------------------------  -------------------------------------------------------------------------------------------------------  
JetCorrectorParametersCollection_Autumn18_RunABCD_V8_DATA_AK4PF       Run       JetCorrectorParametersCollection  any              -1             2019-03-06 17:25:02.887315  Created copying tag JetCorrectorParametersCollection_Autumn18_RunA_V8_DATA_AK4PF from sqlite_file:Au...  
JetCorrectorParametersCollection_Autumn18_RunABCD_V8_DATA_AK4PFchs    Run       JetCorrectorParametersCollection  any              -1             2019-03-06 17:27:18.978273  Created copying tag JetCorrectorParametersCollection_Autumn18_RunA_V8_DATA_AK4PFchs from sqlite_file...  
JetCorrectorParametersCollection_Autumn18_RunABCD_V8_DATA_AK4PFPuppi  Run       JetCorrectorParametersCollection  any              -1             2019-03-06 17:27:19.896530  Created copying tag JetCorrectorParametersCollection_Autumn18_RunA_V8_DATA_AK4PFPuppi from sqlite_fi...  
JetCorrectorParametersCollection_Autumn18_RunABCD_V8_DATA_AK4JPT      Run       JetCorrectorParametersCollection  any              -1             2019-03-06 17:27:20.856122  Created copying tag JetCorrectorParametersCollection_Autumn18_RunA_V8_DATA_AK4JPT from sqlite_file:A...  
JetCorrectorParametersCollection_Autumn18_RunABCD_V8_DATA_AK4Calo     Run       JetCorrectorParametersCollection  any              -1             2019-03-06 17:27:21.770011  Created copying tag JetCorrectorParametersCollection_Autumn18_RunA_V8_DATA_AK4Calo from sqlite_file:...  
JetCorrectorParametersCollection_Autumn18_RunABCD_V8_DATA_AK8PF       Run       JetCorrectorParametersCollection  any              -1             2019-03-06 17:27:22.657777  Created copying tag JetCorrectorParametersCollection_Autumn18_RunA_V8_DATA_AK8PF from sqlite_file:Au...  
JetCorrectorParametersCollection_Autumn18_RunABCD_V8_DATA_AK8PFchs    Run       JetCorrectorParametersCollection  any              -1             2019-03-06 17:27:23.675333  Created copying tag JetCorrectorParametersCollection_Autumn18_RunA_V8_DATA_AK8PFchs from sqlite_file...  
JetCorrectorParametersCollection_Autumn18_RunABCD_V8_DATA_AK8PFPuppi  Run       JetCorrectorParametersCollection  any              -1             2019-03-06 17:27:25.348084  Created copying tag JetCorrectorParametersCollection_Autumn18_RunA_V8_DATA_AK8PFPuppi from sqlite_fi...  
```

**Comment by bortigno** [comment]
Testing MC is successful. Testing the data is also successful. 
I had to modify slightly the test_crab_config.py and the crab/templates/EDAnalyzer.py to make them work with the DB file. Tested on test_crab_config.py was also successful using both a data and MC file.

I will merge to the prod-v18.1.X without PR. 
It seems a quite minor modification to me.

**Comment by bortigno** [comment]
Merged directly to prod-v18.1.X: e3808ae
Closing the issue.

**Comment by bortigno** [comment]
#28
